### ISSUE TITLE
Unify `Die` behavior of initial / cloned sprite

### ISSUE THREAD

**Issue by nighca** [goplus/spx#314]
In SPX, there are different implementations of the `Die` method for initial and cloned sprites:

https://github.com/goplus/spx/blob/3df776b555a0532999520609e172f45c1816de49/sprite.go#L483-L494

Ideally, there should be no noticeable differences for game developers, but this is not the case. For example, consider this sprite code:

```go
onStart => {
	for {
		wait 0.1
		println "im alive"
	}
}

onStart => {
	wait 1
	die
}
```

After calling `die`, the loop continues, so `"im alive"` keeps being printed.

In contrast, for a cloned sprite:

```go
onCloned => {
	for {
		wait 0.1
		println "im alive"
	}
}

onCloned => {
	wait 1
	die
}

onStart => {
	clone
}
```

After calling `die`, the loop stops, and `"im alive"` is no longer printed.

Game developers may perform more complex actions than just `println "im alive"`, potentially affecting overall game behavior. Therefore, it is crucial to unify the behavior of `die` for both sprite types.

Please check this out, @xushiwei @JiepengTan.

**Comment by nighca** [comment]
Based on the comment [here](https://github.com/goplus/builder/issues/734#issuecomment-2309217308)

> In Scratch and SPX, the initial instance of a sprite is the prototype (somewhat similar to JavaScript). The prototype instance cannot be destroyed, so the 'die' function behaves differently for prototype instances and cloned instances.

To address compatibility concerns, I suggest:

* Rename `destroy` to `destroyThisClone` for compatibility with Scratch's [`Delete This Clone`](https://en.scratch-wiki.info/wiki/Delete_This_Clone_(block))
* Ensure that `die` always destroys the current instance, regardless of whether it's the initial or a cloned instance
* If we introduce a `destroy` method, which is more low-level than `die`, in the future, it should also destroys current instance, similar to `die`

**Comment by nighca** [comment]
closed by #329
